client/src/components/content/ModelerSidebar/PropertiesPanel/PropertiesPanelComponents/PPParameterInput.jsx

```diff
@@ -1,43 +1,108 @@
-import React from 'react';
-import { Input, Tooltip } from 'antd';
+import React, { useState } from 'react';
+import { Input, Tooltip, Checkbox } from 'antd';
 import { InfoCircleOutlined } from '@ant-design/icons';
-import PropTypes from 'prop-types'
+import PropTypes from 'prop-types';
 import corporateDesign from '../../../../../layout/corporateDesign';
+import styles from '../../ModelerSidebar.module.css';
+import {
+  parameterPropertyStatus,
+  setPropertyForParameter,
+} from '../../../../../utils/attributeAndParamUtils';
 
 /**
  * @description Renders a parameter input field for a given variable
  * @category Client
  * @component
  */
 const PPParameterInput = ({
-    onValueChange,
-    variableName,
-    isRequired,
-    // eslint-disable-next-line no-unused-vars
-    dataType,
-    value
-}) => (
+  onValueChange,
+  variableName,
+  isRequired,
+  // eslint-disable-next-line no-unused-vars
+  dataType,
+  value,
+  robotId,
+  selectedActivity,
+}) => {
+  const [userInputRequired, setUserInputRequired] = useState(
+    parameterPropertyStatus(
+      robotId,
+      selectedActivity,
+      variableName,
+      'requireUserInput'
+    )
+  );
+  const [parameterValue, setParameterValue] = useState(
+    parameterPropertyStatus(robotId, selectedActivity, variableName, 'value')
+  );
+
+  const changeUserInputRequirement = (event, parameterName) => {
+    if (event.target.checked) {
+      setPropertyForParameter(
+        selectedActivity,
+        parameterName,
+        'requireUserInput',
+        true
+      );
+      setUserInputRequired(true);
+    } else {
+      setPropertyForParameter(
+        selectedActivity,
+        parameterName,
+        'requireUserInput',
+        false
+      );
+      setUserInputRequired(false);
+    }
+  };
+
+  const changeParamterValue = (event, parameterName) => {
+    setPropertyForParameter(
+      selectedActivity,
+      parameterName,
+      'value',
+      event.target.value
+    );
+    setParameterValue(event.target.value);
+  };
+  return (
     <>
-        <Input
-            placeholder={variableName}
-            defaultValue={value}
-            onPressEnter={onValueChange}
-            suffix={
-                isRequired &&
-                <Tooltip title='This field is required for the task to work'>
-                    <InfoCircleOutlined style={{ color: corporateDesign.colorBackgroundCta }} />
-                </Tooltip>
-            }
-        />
+      <Input
+        placeholder={variableName}
+        defaultValue={value}
+        value={userInputRequired ? '' : parameterValue}
+        onChange={(event) => changeParamterValue(event, variableName)}
```

Comment:
Why is this needed? Previously the input needed to be confirmed through enter. Either one therefore should go. 
I have no idea, but how could possibly the performance be influenced by calling the method on every single change? Is it for certain, that the state is updated right this way or could race conditions happen?

Reply:
I always thought that needing to press enter to confirm an input was just a temporary solution and here it was a convenient time for me to just fix that. The performance shouldn't be a problem as we are not working with tha backend. I haven't experienced any kind of racing conditions yet and using state like this to update a value in an input field seems to be (according to my google search and the udemy react course) a common solution :)